mountpoint-s3/src/inode.rs

```diff
@@ -119,9 +119,18 @@ impl Superblock {
             return Err(InodeError::InvalidFileName(name.into()));
         }
 
-        // TODO use caches. if we already know about this name, we just need to revalidate the stat
-        // cache and then read it.
+        let remote = self.remote_lookup(client, parent_ino, name).await?;
+        self.inner.update_from_remote(parent_ino, name, remote)
+    }
 
+    /// Lookup an inode in the parent directory with the given name
+    /// on the remote client.
+    async fn remote_lookup<OC: ObjectClient>(
```

Comment:
Will there ever be a case where you don't immediately call `self.inner.update_from_remote` with the result of `remote_lookup`? If not, maybe we should push the `update_from_remote` call into it so (hypothetical future) callers can't forget.

Reply:
My preference would still be to keep them separate:
* `remote_lookup` wraps all the (double) request logic and returns `Option<RemoteLookup>`,
* `update_from_remote` takes it and updates the local "cache".